packages/tracing/src/utils.ts

```diff
@@ -41,9 +72,56 @@ export function extractTraceparentData(traceparent: string): TraceparentData | u
       parentSpanId: matches[2],
     };
   }
+
   return undefined;
 }
 
+type TracestateHeaderData = { sentry?: string; thirdparty?: string };
+
+/**
+ * Extract data from an incoming `tracestate` header
+ *
+ * @param header
+ * @returns Object containing data from the header
+ */
+export function extractTracestateData(header: string): TracestateHeaderData {
+  let sentryEntry, thirdPartyEntry;
+
+  if (header) {
+    let before, after;
+
+    // find sentry's entry, if any
+    const sentryMatch = SENTRY_TRACESTATE_ENTRY_REGEX.exec(header);
+
+    if (sentryMatch !== null) {
+      sentryEntry = sentryMatch[1];
+
+      // remove the commas after the split so we don't end up with `xxx=yyy,,zzz=qqq` (double commas) when we put them
+      // back together
+      [before, after] = header.split(sentryEntry).map(s => s.replace(/^,*|,*$/g, ''));
+
+      // extract sentry's value from its entry and test to make sure it's valid; if it isn't, discard the entire entry
+      // so that a new one will be created by the Transaction contrcutor
+      const sentryValue = sentryEntry.replace('sentry=', '');
+      if (!new RegExp(`^${base64Stripped}$`).test(sentryValue)) {
```

Comment:
RegExp that doesnt use `g` flag are safe to be initialized once when used with `test` methods.

Reply:
Fixed.